Electron/src/Core/components/Three.js

```diff
@@ -0,0 +1,402 @@
+    let text
+    geometry = new THREE.Geometry()
+    patternFace = /facet([\s\S]*?)endfacet/g
+
+    while ((result = patternFace.exec(data)) !== null) {
+      text = result[0]
+      patternNormal = /normal[\s]+([\-+]?[0-9]+\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]+([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]+([eE][\-+]?[0-9]+)?)+/g
+
+      while ((result = patternNormal.exec(text)) !== null) {
+        normal = new THREE.Vector3(parseFloat(result[1]), parseFloat(result[3]), parseFloat(result[5]))
+      }
+
+      patternVertex = /vertex[\s]+([\-+]?[0-9]+\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]+([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]+([eE][\-+]?[0-9]+)?)+/g
+
+      while ((result = patternVertex.exec(text)) !== null) {
+        geometry.vertices.push(new THREE.Vector3(parseFloat(result[1]), parseFloat(result[3]), parseFloat(result[5])))
+      }
+
+      length = geometry.vertices.length
+
+      geometry.faces.push(new THREE.Face3(length - 3, length - 2, length - 1, normal))
+    }
+
+    geometry.computeBoundingBox()
+    geometry.computeBoundingSphere()
+
+    return geometry
+  },
+
+  ensureString(buf) {
+    if (typeof buf !== "string") {
+      const array_buffer = new Uint8Array(buf)
+      let str = ""
+      for (let i = 0; i < buf.byteLength; i++) {
+        str += String.fromCharCode(array_buffer[i]) // implicitly assumes little-endian
+      }
+      return str
+    } else {
+      return buf
+    }
+  },
+
+  ensureBinary(buf) {
+    if (typeof buf === "string") {
+      const array_buffer = new Uint8Array(buf.length)
+      for (let i = 0; i < buf.length; i++) {
+        array_buffer[i] = buf.charCodeAt(i) & 0xff // implicitly assumes little-endian
+      }
+      return array_buffer.buffer || array_buffer
+    } else {
+      return buf
+    }
+  },
+}
+
+if (typeof DataView === "undefined") {
+  DataView = function (buffer, byteOffset, byteLength) {
+    this.buffer = buffer
+    this.byteOffset = byteOffset || 0
+    this.byteLength = byteLength || buffer.byteLength || buffer.length
+    this._isString = typeof buffer === "string"
+  }
+
+  DataView.prototype = {
+    _getCharCodes(buffer, start, length) {
+      start = start || 0
+      length = length || buffer.length
+      const end = start + length
+      const codes = []
+      for (let i = start; i < end; i++) {
+        codes.push(buffer.charCodeAt(i) & 0xff)
+      }
+      return codes
+    },
+
+    _getBytes(length, byteOffset, littleEndian) {
+      let result
+
+      // Handle the lack of endianness
+      if (littleEndian === undefined) {
+        littleEndian = this._littleEndian
+      }
+
+      // Handle the lack of byteOffset
+      if (byteOffset === undefined) {
+        byteOffset = this.byteOffset
+      } else {
+        byteOffset = this.byteOffset + byteOffset
+      }
+
+      if (length === undefined) {
+        length = this.byteLength - byteOffset
+      }
+
+      // Error Checking
```

Comment:
why multiple if's instead of if/else's?

Reply:
Again, this file was imported and I didn't write directly, but I imagine its because multiple errors could be caught / the errors aren't really related to each other